packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -263,15 +264,28 @@ export class RunningSummarizer implements IDisposable {
                 { refreshLatestAck: true, fullTree: false, delayMinutes: 2 },
                 { refreshLatestAck: true, fullTree: true, delayMinutes: 10 },
             ];
+            let overrideDelaySeconds: number | undefined;
+            let retryAttempt = 0;
             for (let i = 0; i < attempts.length; i++) {
                 const { delayMinutes = 0, ...options } = attempts[i];
-                if (delayMinutes > 0) {
-                    await new Promise((resolve) => setTimeout(resolve, delayMinutes * 1000 * 60));
+                const delaySeconds = overrideDelaySeconds ?? (delayMinutes * 60);
+                if (delaySeconds > 0) {
+                    await new Promise((resolve) => setTimeout(resolve, delaySeconds * 1000));
                 }
-                const attemptReason = i > 0 ? `retry${i}` as `retry${number}` : reason;
-                if (await this.generator.summarize(attemptReason, options) === true) {
+                const attemptReason = retryAttempt > 0 ? `retry${retryAttempt}` as `retry${number}` : reason;
```

Comment:
From your description, it feels like we have retry 1.1, 1.2, 2.1, 2.2, 3.1 & 3.2.
I.e. we have phase (that describes qualities like full, refresh) and also up to two retries per phase.
Do we have enough telemetry to record both phase & retry?

Reply:
We have fullTree and refreshLatestAck as properties in the event, so we should be good in that way. We also have reason there, which now can be parallel to fullTree/refreshLatestAck.

The only part we can't deduce is the delay (if any). I've added a separate telemetry event for that. Let me know if you think that's too much @vladsud

Reply:
Sounds good. Working with actual telemetry data will provide further input on if we want it in different form :)